src/Raven.Server/Documents/Handlers/RevisionsHandler.cs

```diff
@@ -61,7 +62,29 @@ public Task GetRevisionsConfig()
         [RavenAction("/databases/*/admin/revisions/config", "POST", AuthorizationStatus.DatabaseAdmin)]
         public async Task ConfigRevisions()
         {
-            await DatabaseConfigurations(ServerStore.ModifyDatabaseRevisions, "read-revisions-config");
+            await DatabaseConfigurations(
+                ServerStore.ModifyDatabaseRevisions,
+                "read-revisions-config",
+                beforeSetupConfiguration: (string name, ref BlittableJsonReaderObject configuration, JsonOperationContext context) =>
```

Comment:
I'm assuming that this happens before we go to the cluster state machine, right?

Reply:
What happens if we have two separate requests like this? Is this a whole sale replacement over time, or individual ops?

Reply:
Full replacement: https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Server/ServerWide/Commands/EditRevisionsConfigurationCommand.cs#L28

Reply:
yes,  this happens before we go to the cluster state machine